sys/syz-extract/extract.go

```diff
@@ -25,6 +25,7 @@ var (
 	flagBuild     = flag.Bool("build", false, "regenerate arch-specific kernel headers")
 	flagSourceDir = flag.String("sourcedir", "", "path to kernel source checkout dir")
 	flagIncludes  = flag.String("includedirs", "", "path to other kernel source include dirs separated by commas")
+	flagDefines   = flag.String("defines", "", "macro defines separated by commas")
```

Comment:
You want to control the defines that the included headers see, right?
Because otherwise we do have the `define`-related functionality: `grep -r define ./sys/linux/`

I wonder if it's possible to reuse it for this particular purpose as well. E.g. if some `define` comes before `include <>` in a descriptions file, then we should also put that `#define` before `#include`s during C code generation in `fetch.go`.

Reply:
Presumably you have some build errors w/o the defines, so I wonder how these headers are used/built in the actual kernel build. We also provide real kernel config (and presumably a real config with the new flag), yet headers of your kernel does not build. Maybe you need to include some other header before the broken header?
e.g.:
```
include <should_be_included_before_broken_foo.h>
include <broken_foo.h>
```

Reply:
you are right, no need defines if I specify -config now.

Reply:
So do you still need the `sys/syz-extract: add -defines flag to pass macro defines to cc` commit?

Reply:
Removed it now as -config works for normal kernel build.